shared/src/commonMain/kotlin/com/example/tasky/database/model/EventEntity.kt

```diff
@@ -0,0 +1,66 @@
+package com.example.tasky.database.model
+
+import androidx.room.Embedded
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.Index
+import androidx.room.PrimaryKey
+import androidx.room.Relation
+import com.example.tasky.agenda.domain.model.RemindAtType
+
+data class EventWithAttendeeAndPhoto(
+    @Embedded val eventEntity: EventEntity,
+    @Relation(parentColumn = "id", entityColumn = "eventId")
+    val attendees: List<AttendeeEntity>,
+    @Relation(parentColumn = "id", entityColumn = "eventId")
+    val photos: List<PhotoEntity>,
+)
+
+@Entity
+data class EventEntity(
```

Comment:
I also considered splitting up the entities like this, but I found a single table was sufficient. Feel free to keep yours as is. I will share an alternative with you so that you are aware of it.

```
@Entity
data class EventEntity(
  @PrimaryKey(autoGenerate = false)
  val id: String,
  val title: String,
  val description: String?,
  val hostId: String,
  val isUserEventCreator: Boolean,
  val fromTime: Long,
  val toTime: Long,
  val remindBefore: RemindBeforeDuration,
  val photos: List<PhotoSerialized>,
  val attendees: List<AttendeeSerialized>,
)
```

```
@Serializable
data class AttendeeSerialized(
  val userId: String,
  val email: String,
  val name: String,
  val eventId: String,
  val isGoing: Boolean,
  val remindAt: Long,
  val isCreator: Boolean,
)
```

```
@Serializable
data class PhotoSerialized(
  val key: String,
  val url: String
)
```

Reply:
what would be saved in the room database if it is not entity?

Reply:
It will be an entity, but a single entity is a single table, not multiple entities (each with its own table). The approach I shared is more straightforward. Simpler is not always the answer; sometimes you need to have multiple tables, similar to your approach, but I tend to opt for the simpler approach when possible 😄